src/lib/Car/BeastCar.ts

```diff
@@ -1,90 +1,66 @@
+import calculateBeastPrice from "../../helpers/Calculators/CalculateBeastPrice";
 import type BaseCar from "./BaseCar";
-import { breakdownMinutes } from "../Time";
 
 import BeastLogo from '$lib/Images/beast.png';
-import type { SvelteComponent } from "svelte";
+import type {SvelteComponent} from "svelte";
 import BeastCarPopover from "$lib/Popovers/BeastCarPopover.svelte";
-
-type BeastCarObject = {
-  name: string,
-  price: {
-    km: number,
-    minute: number,
-    day: number,
-    threeDays: number,
-    week: number,
-    start: number,
-    deposit: {
-      base: number,
-      special: number,
-    },
-  },
-}
+import type {SearchParamsObj} from "../DTO/SearchParamsObj";
+import type {Car} from "../DTO/Car";
 
 class BeastCar implements BaseCar {
 
-  readonly carData: BeastCarObject;
+  readonly carData;
+  rentTotalPrice: number | undefined;
 
   /**
    * Initialise car data.
-   * 
+   *
    * @param car The car object.
    */
-  constructor( car: BeastCarObject ) {
-    
+  constructor( car: Car ) {
+
     this.carData = car;
   }
 
+
   /**
    * @inheritdoc
    */
   getName(): string {
-    
+
     return this.carData.name;
   }
-  
+
   /**
    * @inheritdoc
    */
-  getTotalPrice( minutes: number, distance: number ): number {
-
-    let duration       = breakdownMinutes( minutes );
-    let kilometrePrice = 0;
-    let durationPrice  = 0;
-
-    if ( distance > ( 300 + duration.days*300 ) ) {
-      kilometrePrice = this.carData.price.km;
-    }
-
-    let theeDayCount = Math.floor( duration.days/3 );
-    let weekCount    = Math.floor( duration.days/7 );
-
-    durationPrice = Math.min( 
-      duration.minutes * this.carData.price.minute + (duration.hours*60) * this.carData.price.minute + duration.days * this.carData.price.day, 
-      this.carData.price.day + duration.days*this.carData.price.day,
-      this.carData.price.threeDays + theeDayCount*this.carData.price.threeDays,
-      this.carData.price.week + weekCount*this.carData.price.week,
-    );
-
-    let total = durationPrice + distance*kilometrePrice;
+  calculateRentTotalPrice(searchParamsObj: SearchParamsObj): void {
+    this.rentTotalPrice = calculateBeastPrice(this.carData, JSON.parse(JSON.stringify(searchParamsObj)))
```

Comment:
`JSON.parse(JSON.stringify(searchParamsObj))` feels like a shortcut, why is it needed like this?

Reply:
Deep copy is necessary here because some calculation change some search parameters. Fro example: src/helpers/Calculators/CalculateBeastPrice.ts line 23. I don't also like that but it is the first solution that came to my mind.